src/cpp/cpp_constructor.cpp

```diff
@@ -235,8 +235,8 @@ optionalt<codet> cpp_typecheckt::cpp_constructor(
     assert(initializer.id()==ID_code &&
            initializer.get(ID_statement)==ID_expression);
 
-    side_effect_expr_function_callt &func_ini=
-      to_side_effect_expr_function_call(initializer.op0());
+    side_effect_expr_function_callt &func_ini =
+      to_side_effect_expr_function_call(to_unary_expr(initializer).op());
```

Comment:
This seems wrong: we above assert that it's a statement expression.

Reply:
The side_effect_expr_function_call is the operand of the statement expression.

Reply:
Ah, sorry, my comment was unclear: I think using `to_unary_expr` on something that actually is a `codet` seems wrong (even when the type system currently permits it).

Reply:
Now using `side_effect_expr_statement_expressiont`